Sveriges radio.js

```diff
@@ -0,0 +1,709 @@
+		return "multiple";
+	}
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// TODO: adjust the xpath
+	var rows = doc.querySelectorAll('h2>a.title[href*="/article/"]');
+	for (let i=0; i<rows.length; i++) {
+		// TODO: check and maybe adjust
+		let href = rows[i].href;
+		// TODO: check and maybe adjust
+		let title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		item.publicationTitle = "Sveriges Radio";
+
+		item.creators = [];
+		var nameNodes = ZU.xpath(doc, '//p[@class="byline"]/text()');
+		for(var node of nameNodes) {
+			var nameString = node.wholeText;
+			if(nameString.trim() === "") {
+				// Empty text node are added between authors; we want to skip
+				// those.
+				continue;
+			}
+			var names = nameString.split(" ");
+			var firstName = names[0];
+			var lastName = names[1];
+			if(lastName === undefined) {
+				lastName = "";
+			}
+			var author = {
+				// Assume only one first name.
+				firstName: firstName,
+				lastName: lastName,
+				creatorType: "author"
+			};
+			item.creators.push(author);
+		}
+
+		// The title from the meta is in the format:
+		// Australierna säger ja till samkönade äktenskap - Nyheter (Ekot)
+		var titleString =
+			ZU.xpath(doc, '//meta[@name="twitter:title"]')[0]
+			.getAttribute("content");
+		var titleParts = titleString.split(" - ");
+		item.title = titleParts[0];
+		item.section = titleParts[1];
+
+		var dateString = 
+			ZU.xpath(doc, '//meta[@name="displaydate"]')[0]
+			.getAttribute("content");
+		item.date = dateString.replace(/(....)(..)(..)/, "$1-$2-$3");
+
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "newspaperArticle";
+		// TODO map additional meta tags here, or delete completely
+		trans.addCustomFields({
+			'twitter:description': 'abstractNote'
+		});
+		trans.doWeb(doc, url);
+	});
+}
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=6821850",
+		"items": [
+			{
+				"itemType": "newspaperArticle",
+				"title": "Australierna säger ja till samkönade äktenskap",
+				"creators": [
+					{
+						"firstName": "Ekot",
```

Comment:
using Zotero's built in function will solve this, but the right way to handle single-name and institutional authors is to use `lastName` only and set `fieldMode: true`